ports/libigl/portfile.cmake

```
@@ -1,52 +1,51 @@
+# Header-only library
+
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO libigl/libigl
-    REF  e60423e28c86b6aa2a3f6eb0112e8fd881f96777 #2.3.0
-    SHA512 3fecb710825e58745c1d67eab694ee365a5b86151a5a1ca3758c1000c124059d38dbc78e8c6e941be6d85a716f928ed8fea42bb6007b8e24da0123332c2c96da
+    REF  66b3ef2253e765d0ce0db74cec91bd706e5ba176 #2.4.0
+    SHA512 7014ffdaa160bfa2509fc283cb7176d7994a37f51509c7374659292efad076c8fb594f9f6990bab1aa5562d1f66e93403ea35a5bf2a924436560a2d4669ffcfd
     HEAD_REF master
     PATCHES
-        fix-dependency.patch
-        install-extra-headers.patch
-        fix-config.patch
+        dependencies.patch
 )
 
 vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
     FEATURES
-        embree LIBIGL_WITH_EMBREE
-        opengl LIBIGL_WITH_OPENGL
-        glfw   LIBIGL_WITH_OPENGL_GLFW
-        imgui  LIBIGL_WITH_OPENGL_GLFW_IMGUI
-        #png LIBIGL_WITH_PNG # Disable this feature due to issue https://github.com/libigl/libigl/issues/1199
-        xml    LIBIGL_WITH_XML
-        #python LIBIGL_WITH_PYTHON # Python binding are in the process of being redone.
-        cgal   LIBIGL_WITH_CGAL
+        embree          LIBIGL_EMBREE
+        opengl          LIBIGL_OPENGL
+        glfw            LIBIGL_GLFW
+        imgui           LIBIGL_IMGUI
+        png             LIBIGL_PNG
+        xml             LIBIGL_XML
+        cgal            LIBIGL_COPYLEFT_CGAL
+        predicates      LIBIGL_PREDICATES
 )
 
+# remove custom FindGMP and FildMPFR
+file(REMOVE "${SOURCE_PATH}/cmake/find/FindGMP.cmake")
+file(REMOVE "${SOURCE_PATH}/cmake/find/FindMPFR.cmake")
+
 vcpkg_cmake_configure(
     SOURCE_PATH "${SOURCE_PATH}"
     DISABLE_PARALLEL_CONFIGURE
     OPTIONS ${FEATURE_OPTIONS}
-        -DLIBIGL_BUILD_PYTHON=OFF
-        -DLIBIGL_EXPORT_TARGETS=ON
-        -DLIBIGL_USE_STATIC_LIBRARY=OFF # Header-only mode
-        -DLIBIGL_WITH_COMISO=OFF
-        -DLIBIGL_WITH_TETGEN=OFF
-        -DLIBIGL_WITH_TRIANGLE=OFF
-        -DLIBIGL_WITH_PREDICATES=OFF
-        -DLIBIGL_BUILD_TUTORIALS=OFF
-        -DLIBIGL_WITH_PNG=OFF
+        # Build options
         -DLIBIGL_BUILD_TESTS=OFF
-        -DPYTHON_EXECUTABLE=${PYTHON2}
-    MAYBE_UNUSED_VARIABLES
-        PYTHON_EXECUTABLE
-)
+        -DLIBIGL_BUILD_TUTORIALS=OFF
+        -DLIBIGL_INSTALL=ON
+        -DLIBIGL_USE_STATIC_LIBRARY=OFF
+        -DHUNTER_ENABLED=OFF
+        -DLIBIGL_COPYLEFT_COMISO=OFF #there is no comiso port available anywhere. solved internally via FetchContent in cmake\recipes\external\comiso.cmake. maybe replace in patch with vcpkg_from_github 
+        -DLIBIGL_COPYLEFT_TETGEN=OFF #there is no tetgen port available anywhere. solved internally via FetchContent in cmake\recipes\external\tetgen.cmake
+        -DLIBIGL_RESTRICTED_TRIANGLE=OFF #same
+        )
 
 vcpkg_cmake_install()
-
-vcpkg_cmake_config_fixup(CONFIG_PATH share/libigl/cmake)
+vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake)
```

```suggestion
vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/libigl PACKAGE_NAME libigl)
```
Your config stuff doesn't work

---

the config is also not including the target file.....

---

Considering previous ports (e.g. https://github.com/microsoft/vcpkg/blob/0bf3923f9fab4001c00f0f429682a0853b5749e0/ports/libigl/portfile.cmake#L68) I would leave it unchanged